projects/plugins/crm/includes/ZeroBSCRM.GeneralFuncs.php

```diff
@@ -1553,4 +1553,61 @@ function jpcrm_youtube_url_to_video_id( $video_url ) {
 
 /* ======================================================
    / YouTube Helpers
-   ====================================================== */
\ No newline at end of file
+====================================================== */
+
+// These constants were created to simplify the process when this page gets
+// moved to a React-based approach. It was a choice to not refactor any further
+// than this.
+define( 'SETTING_SECOND_ADDRESS_ALWAYS_HIDE', 0 );
+define( 'SETTING_SECOND_ADDRESS_ALWAYS_SHOW', 1 );
+define( 'SETTING_SECOND_ADDRESS_ALLOW_CHANGE_VISIBILITY', 2 );
+
+/**
+ * This function returns the correct value for the second address visibility
+ * based on two distinct settings keys. We are using two different settings keys
+ * because the option to allow changing the visibility was added later, and we
+ * want to keep backwards compatibility. After all uses of the second address
+ * visibility are dealt with (e.g. in the Client Portal), a migration
+ * can be written to use only one setting key.
+ *
+ * @param array $settings The settings array.
+ * @return int The value for the visibility of the second address fields.
+ */
+function jpcrm_settings_get_second_address_visibility( $settings ) {
+	if ( ! isset( $settings['secondaddress'] ) || $settings['secondaddress'] !== 1 ) {
+		return SETTING_SECOND_ADDRESS_ALWAYS_HIDE;
+	}
+
+	if ( isset( $settings['secondaddresstoggle'] ) && $settings['secondaddresstoggle'] === 1 ) {
+		return SETTING_SECOND_ADDRESS_ALLOW_CHANGE_VISIBILITY;
+	}
+
+	return SETTING_SECOND_ADDRESS_ALWAYS_SHOW;
+}
+
+/**
+ * This function returns the correct key for a field.
+ * We have to use this because some fields changed their keys (i.e. second
+ * address field) and their usage was not correctly changed/migrated.
+ * This is used for backwards compatibility only.
+ *
+ * @param string $field_key The field key.
+ * @return string The modified field key if it needs to be changed, otherwise returns the same $field_key.
+ */
+function jpcrm_compatibility_get_correct_object_key( $field_key ) {
```

Comment:
(Purely personal preference)

```suggestion
function jpcrm_convert_legacy_object_key( $field_key ) {
```

Reply:
Addressed in ed43899.